merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -74,44 +81,41 @@ export class Windows {
   }
 
   /**
-   * Produce a window when all arguments produce a window.
-   *
-   * Performs the intersection of the argument windows.
+   * Performs the boolean And operation on any number of Windows.
    *
    * @param windows any number of windows expressions
    */
-  public static All(...windows: Windows[]): Windows {
+  public static And(...windows: Windows[]): Windows {
     return new Windows({
-      kind: AST.NodeKind.WindowsExpressionAll,
+      kind: AST.NodeKind.WindowsExpressionAnd,
       expressions: [...windows.map(other => other.__astNode)],
     });
   }
 
   /**
-   * Produce a window when any argument produces a window.
+   * Performs the boolean Or operation on any number of Windows.
    *
-   * Performs the union of the argument windows.
-   *
-   * @param windows one or more windows expressions
+   * @param windows any number of windows expressions
```

Comment:
Can we call this with 0 windows expressions?

Reply:
Yes. `Windows.Or()` returns false and `Windows.And()` return true. Which is kinda vacuously correct. None of the arguments to Or were true, and all of the arguments to And were false. 🤷

Reply:
Okay, just to confirm, when all the arguments to `Or` are false, `Or` returns false, and when all of the arguments to `And` are false, `And` returns true?

Reply:
whoops, typo: `Windows.And()` returns true because all of the arguments were _true_

Reply:
Gotcha. And calling `Or` without arguments means all arguments are false, while calling `And` with no arguments means all arguments are true?

Reply:
Yeah. `Or` returns true if any of the arguments are true, and `And` returns false if any of the arguments are false. Even with no arguments this is still technically correct, which is the best kind of correct